docs/developer/releasing.md

```diff
@@ -62,26 +62,26 @@ export GPG_TTY=$(tty)
 
 ## Performing Releases
 
-The lifetime of a release branch is sheparded by a single person, picked from a
-rota of project maintainers and contributors. The release shepard is
+The lifetime of a release branch is shepherded by a single person, picked from a
+rota of project maintainers and contributors. The release shepherd is
 responsible for managing all releases that compose the release branch.
 
-For a new release branch, the release shepard will:
+For a **new release branch**, the release shepherd will:
 
 1. Gather consensus on which commit should be used as a base for the release
    branch.
 
-2. Create and push the release branch from the selected base commit.
+2. Create and push the release branch from the selected base commit. 
+
+   The name of the release branch should be the name of the stable version we intend 
+   to release, e.g. `release-v0.31`. Release branches do not contain the `-rc.N` release 
+   candidate suffix, i.e. there is no `release-v0.31-rc.0`.
 
 3. Create a PR to cherry-pick additional commits into the release branch as
    needed.
 
-4. Create a PR to [update code](#updating-code) for the upcoming release
-   candidate. A new section in the changelog should be added for the release
-   candidate, documenting all changes introduced by the release candidate.
-
-4. Update the changelog with a new section for the upcoming release candidate,
-   documenting changes that will be introduced in that release candidate.
+4. Create a PR to [update code](#updating-code) using the `-rc.N` release
+   candidate version.
 
 5. Tag the release candidate using the tag naming convention `vX.Y.0-rc.N`.
```

Comment:
I'm worried that the instructions here might be a little confusing. We should publish the release candidate too, but the instructions don't mention it. 

Maybe there should be a dedicated section for post-publishing steps, where:

For a release candidate:

* Publish the release candidate 
* Announce the release candidate in the public Slack channel 

For a stable release:

* Publish the release candidate 
* Ensure Homebrew gets updated 
* Ensure documentation got published 
* Announce the stable release in the public Slack channel

Reply:
True, it does get confusing. I was trying to think how to implement the  post-publishing steps suggestion, but couldn't think of a super clean way.

For now I split the entire "candidate" and the "stable" steps into two different lists. I feel like this makes sense, since they are usually done on two different days? That way we start from point 1 on the second day.